2020/day4/solution.py

```diff
@@ -0,0 +1,57 @@
+import re
+kvp_matcher = re.compile(r"(\S+):(\S+)")
+
+class Passport:
+    RequiredPropertyValidatorMap = {
+        'byr' : lambda x: int(x) >= 1920 and int(x) <= 2002,
+        'iyr' : lambda x: int(x) >= 2010 and int(x) <= 2020,
+        'eyr' : lambda x: int(x) >= 2020 and int(x) <= 2030,
+        'hgt' : lambda x: (len(x) == 4 or len(x) == 5) and ((x[-2:] == "in" and int(x[:-2]) >= 59 and int(x[:-2]) <= 76) or (x[-2:] == "cm" and int(x[:-2]) >= 150 and int(x[:-2]) <= 193)),
+        'hcl' : lambda x: len(x) == 7 and x[0] == "#" and all(map(lambda y: y in "0123456789abcdef" , x[1:])),
+        'ecl' : lambda x: x in ['amb', 'blu', 'brn', 'gry', 'grn', 'hzl', 'oth'],
+        'pid' : lambda x: len(x) == 9 and all(map(lambda y: y in "0123456789" , x)),
+    }
+
+    OptionalProperties = [
+        'cid',
+    ]
+
+    def __init__(self):
+        self._properties = {}
+
+    def addLine(self, line):
+        kvpairs = map(lambda x: x.strip(), line.split(" "))
```

Comment:
`line.split()` (no arguments) automagically does the right thing!  - https://docs.python.org/3/library/stdtypes.html#str.split

Reply:
that's good,  more robust than what i have